product_manufacturer/hooks.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2022 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.tools import sql
+
+
+def pre_init_hook(cr):
+    table_name = "product_template"
+    for field_name, field_type in [
+        ("manufacturer", "int4"),
+        ("manufacturer_pname", "varchar"),
+        ("manufacturer_pref", "varchar"),
+        ("manufacturer_purl", "varchar"),
+    ]:
+        if sql.column_exists(cr, table_name, field_name):
+            continue
+        sql.create_column(cr, table_name, field_name, field_type)
```

Comment:
@mt-software-de Do you have any metrics about that as this involves no data? So, why not relying on ORM for that (this would imply we would do that in very module...)?

Reply:
in our case we have a database with 
- 361645 records in product_product
- 355095 in product_template

i didn't measured the time, but in fact on installing this module there will be no data in this columns on product_product. 
Which means the method _compute_manufacturer_info will result in empty values for all of the product.template's fields.

Reply:
i updated the comment of the PR maybe it is now more understandable? :-)

Reply:
Yes, but the problem is that for existing data, you don't fill in the missing data in product_template table.

So, this is not sufficent and not acceptable as is (as no data will be computed at all).

Reply:
oh yes this of course a problem.